js/bibox.js

```diff
@@ -1007,6 +1008,102 @@ module.exports = class bibox extends Exchange {
         return this.parseBalance (response);
     }
 
+    parseLedgerEntry (item, currency = undefined) {
+        //
+        //    {
+        //        "i": 1125899918063693495,     // entry id
+        //        "s": "USDT",                  // asset symbol
+        //        "T": "transfer_in",           // entry type: transfer, trade, fee
+        //        "a": 14.71,                   // amount
+        //        "b": 14.7100000044,           // balance
+        //        "t": 1663367640374            // time
+        //    }
+        //
+        const ledgerTypes = {
+            'transfer_in': 'transfer',
+            'transfer_out': 'transfer',
+            'trade_finish_ask': 'trade',
+            'trade_finish_bid': 'trade',
+        };
+        const id = this.safeString (item, 'i');
+        const currencyId = this.safeString (item, 's');
+        const type = this.safeString (item, 'T');
+        const timestamp = this.safeInteger (item, 't');
+        const amount = this.safeString (item, 'a');
+        let direction = 'in';
+        if (Precise.stringLt (amount, '0')) {
+            direction = 'out';
+        }
+        return {
+            'id': id,
+            'direction': direction,
+            'account': undefined,
+            'referenceId': id,
+            'referenceAccount': undefined,
+            'type': ledgerTypes[type],
```

Comment:
@samgermain please avoid direct accesses, use `safeString` instead
![image](https://user-images.githubusercontent.com/43336371/197480868-6db2a030-406f-4575-a893-7c689c6fef76.png)
 
moreover, can you fix this crash? (T is fee and since it's not inside ledgerTypes crashes) ⬆️